Dockerfile

```diff
@@ -0,0 +1,23 @@
+FROM centos:latest
+
+MAINTAINER Justin Henderson justin@hasecuritysolutions.com
+
+RUN yum update -y
+RUN yum install -y python python-devel git gcc
+RUN curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py
+RUN python get-pip.py
+
+WORKDIR /opt/VulnWhisperer
+
+COPY requirements.txt requirements.txt
```

Comment:
The Docker image should only contain what is needed to run the intended goal of the docker image.   Having .git, docker-compose files, logstash config files, is normally not desired.  Especially for a docker image that will have significant access to a backend security product in a production environment.

I'm not following what you mean about the user having to download a specific version of VulnWhisperer.  Most users will likely either 1) use the docker image in hub.docker.io and either identify the :latest or identify a specific tag they know works in their environment 2) will use Dockerfile to build from scratch.  In that case, they are already checking out the desired version of the code from Git.

Reply:
You are right, I understand your security concerns; I was just more focused on the user case, as having those files even if not desired, it is just few extra space, so it should have very little impact on the container itself. I am no Docker expert at all, so probably your approach is better, just followed @SMAPPER original Dockerfile, which was oriented on having VulnWhisperer working out of the box without any extra dependency, only needing the config file with creds and parameters.

Regarding the specific version, this was something I found out when testing the original Dockerfile; it was not doing the git clone of the repository because at some point, an old version of VulnWhisperer was cached and docker was not checking if it was the latest version of the git repository. Because of that layering issue, it was important to specify the version to make sure you were getting what you expected. 

In your Dockerfile case this is not the case as it will depend on the specific version that was downloaded, but because of the dependency of your Dockerfile that requires to be executed inside of an already cloned VulnWhisperer's root folder, it would not work to users that pulled the image from hub.docker.com directly.